compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaUtils.cpp

```diff
@@ -13,6 +13,31 @@
 
 namespace mlir::iree_compiler::AMDAIE {
 
+static bool isEqualConstantIntOrValueArrayFromIndices(
+    const SmallVector<OpFoldResult> &ofrsA,
+    const SmallVector<OpFoldResult> &ofrsB, size_t indexA = 0,
+    size_t indexB = 0) {
+  if ((ofrsA.size() - indexA) != (ofrsB.size() - indexB)) return false;
+  ArrayRef<OpFoldResult> viewA(ofrsA.begin() + indexA, ofrsA.end());
+  ArrayRef<OpFoldResult> viewB(ofrsB.begin() + indexB, ofrsB.end());
+  return isEqualConstantIntOrValueArray(viewA, viewB);
+}
+
+bool areAccessPatternsEqualFromIndices(
+    const SmallVector<OpFoldResult> &offsetsA,
```

Comment:
nit: do we prefer `ArrayRef<OpFoldResult>` to `const SmallVector<OpFoldResult> &` ? 

With ArrayRef you can use drop_front https://llvm.org/doxygen/classllvm_1_1ArrayRef.html#a55089293ebaccd683a82d97170041376

Reply:
Yes, we should prefer `ArrayRef`, updated it. Thanks for the pointer on `drop_front`.